rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -585,19 +632,22 @@ public NodeSet select(int number, List<Node> matchedNodes) {
             NodeSet result = new NodeSet();
             for (Integer i : sortedCapacities) {
                 for (InetAddress host : hostsMap.get(i)) {
-                    List<Node> hostNodes = nodesOnHost.get(host);
-                    result.add(hostNodes.get(0));
-                    if (hostNodes.size() > 1) {
-                        List<Node> newExtraNodes = new LinkedList<>(hostNodes
-                                .subList(1, hostNodes.size()));
-                        if (result.getExtraNodes() == null) {
-                            result.setExtraNodes(new LinkedList<Node>());
+                    Set<Node> hostNodes = nodesOnHost.get(host);
+                    int sz = hostNodes.size();
+                    if (sz > 0) {
+                        result.add(hostNodes.iterator().next());
```

Comment:
Indeed it is. The problem is that I absolutely don't understand this code, and if I modify it, I will probably break it.